UnityProject/Assets/Scripts/Player/ExtinguishByHand.cs

```diff
@@ -0,0 +1,29 @@
+using UnityEngine;
+
+/// <summary>
+/// Allows an object to be patted when on fire.
+/// </summary>
+public class ExtinguishByHand : MonoBehaviour, IClientInteractable<PositionalHandApply>
+{
+	public bool Interact(PositionalHandApply interaction)
+	{
+		var targetPlayerHealth = interaction.TargetObject.GetComponent<PlayerHealth>();
+		var performerPlayerHealth = interaction.Performer.GetComponent<PlayerHealth>();
+		var performerRegisterPlayer = interaction.Performer.GetComponent<RegisterPlayer>();
+
+		// Is the target in range for a pat? Is the performer's intent set to help?
+		// Is the performer's hand empty? Is the performer not stunned/downed? Is the performer conscious to perform the interaction?
+		if (!PlayerManager.LocalPlayerScript.IsInReach(interaction.WorldPositionTarget, false) ||
+			UIManager.CurrentIntent != Intent.Help ||
+			interaction.HandObject != null ||
+			performerPlayerHealth.ConsciousState != ConsciousState.CONSCIOUS ||
+			performerRegisterPlayer.IsLayingDown)
+		{
+			return false;
+		}
+
+		PlayerManager.LocalPlayerScript.playerNetworkActions.CmdRequestPatFire(PlayerManager.LocalPlayerScript.playerName,
```

Comment:
You don't need to pass the player name into this, playerNetworkActions already lives on the same object that the local player script does. You can just access playerScript from within playerNetworkActions to retrieve player name. The way commands work, the local player is only allowed to call commands on their local player object. PNA lives on the local player object.

Reply:
I see... that makes sense.

Huggable does things in a similar way, which is why I implemented it like that:
```
PlayerManager.LocalPlayerScript.playerNetworkActions.CmdRequestHug(PlayerManager.LocalPlayerScript.playerName,
    gameObject);
```
Would you like me to change Huggable as well?